pkg/chartmuseum/server/multitenant/handlers.go

```diff
@@ -0,0 +1,88 @@
+package multitenant
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/kubernetes-helm/chartmuseum/pkg/repo"
+
+	"github.com/gin-gonic/gin"
+)
+
+var (
+	warningHTML = []byte(`<!DOCTYPE html>
+<html>
+<head>
+<title>WARNING</title>
+</head>
+<body>
+<h1>WARNING</h1>
+<p>This ChartMuseum install is running in multitenancy mode.</p>
+<p>This feature is still a work in progress, and is not considered stable.</p>
+<p>Please run without the --multitenant flag to disable this.</p>
+</body>
+</html>
+	`)
+)
+
+func (server *MultiTenantServer) defaultHandler(c *gin.Context) {
+	c.Data(200, "text/html", warningHTML)
+}
+
+func (server *MultiTenantServer) getIndexFileRequestHandler(c *gin.Context) {
```

Comment:
What do you think about separating the web handler related operations from the actual work?`getIndexFileRequestHandler()` would delegate to let's say: `getIndexFile(orgName, repoName, prefix)`, and this method would contain only the objects processing. This would allow better unittesting in the long run, and cleanly isolate the details of the web framework.

Reply:
I've opened https://github.com/kubernetes-helm/chartmuseum/issues/75 to address this